swap_meet/vendor.py

```diff
@@ -1,2 +1,139 @@
+            other_Vendor.inventory.append(my_item)
+
+            other_Vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, other_Vendor):
+        """
+        Swaps the first item in two vendor's inventory lists and
+        returns True; if either inventory list is empty, method returns
+        False
+        """
+        if self.inventory == [] or other_Vendor.inventory == []:
+            return False
+
+        else:
+            first_item_vendor = self.inventory[0]
+            first_item_other_vendor = other_Vendor.inventory[0]
+
+            if first_item_vendor and first_item_other_vendor:
+                self.inventory.remove(first_item_vendor)
+                other_Vendor.inventory.append(first_item_vendor)
+
+                other_Vendor.inventory.remove(first_item_other_vendor)
+                self.inventory.append(first_item_other_vendor)
+
+                return True
+
+    def get_by_category(self, category):
+        """
+        Returns a list of items in the same category; if none match,
+        returns an empty list
+        """
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+
+        return category_list
+
+    def get_best_by_category(self, category):
+        """
+        Returns the best condition item in a vendor's inventory. If no
+        item matches the given category, method returns None
+        """
+        best_by_category_list = []
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                best_by_category_list.append(item)
+
+        if not best_by_category_list:
+            return None
+
+        best_condition = best_by_category_list[0].condition
+        best_condition_item = best_by_category_list[0]
+
+        for item in best_by_category_list:
+            if item.condition > best_condition:
+                best_condition = item.condition
+                best_condition_item = item
+
+        return best_condition_item
+
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        """
+        Swaps the best item in a given category between two vendor's
+        inventories, based on respective priority, and return True. If
+        either inventory does not have an item in that category,
+        method returns False
+        """
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other_vendor.get_best_by_category(my_priority)
+
+        if not my_best_item or not their_best_item:
+            return False
+
+        self.inventory.remove(my_best_item)
+        self.inventory.append(their_best_item)
+        other_vendor.inventory.remove(their_best_item)
+        other_vendor.inventory.append(my_best_item)
```

Comment:
I know y'all have experience using your instance functions! I wonder if we have one other instance function that will do all of this for us... 🤔